osgp/shared/osgp-dto/src/main/java/org/opensmartgridplatform/dto/valueobjects/smartmetering/SmartMeteringDeviceDto.java

```diff
@@ -11,207 +11,35 @@
 import java.io.Serializable;
 import java.util.Date;
 
+import lombok.Data;
+
+@Data
 public class SmartMeteringDeviceDto implements Serializable {
 
     private static final long serialVersionUID = -6133164707489276802L;
 
     private String deviceIdentification;
-
     private String deviceType;
-
     private String communicationMethod;
-
     private String communicationProvider;
-
     private String iccId;
-
     private String protocolName;
-
     private String protocolVersion;
-
     private byte[] masterKey;
-
     private byte[] globalEncryptionUnicastKey;
-
     private byte[] authenticationKey;
-
     private String supplier;
-
     private boolean hls3Active;
-
     private boolean hls4Active;
-
     private boolean hls5Active;
-
     private Date deliveryDate;
-
     private Long mbusIdentificationNumber;
-
     private String mbusManufacturerIdentification;
-
     private Short mbusVersion;
-
     private Short mbusDeviceTypeIdentification;
-
     private byte[] mbusDefaultKey;
+    private String ipAddress;
+    private int cellId;
+    private int btsId;
```

Comment:
It feels odd that these are included here and not with the `SmartMeteringDevice` which is tightly related.
Could this have to do with an underlying issue that the new fields belong with Device rather than with SmartMeter?

Reply:
I double checked this during development with our PO, should be part of Device (i.e. any CDMA device).

Reply:
I think this is due to the fact that the hierarchy from the entities is not reflected in the value objects.
This way value objects and DTOs for specific sub-types of `Device` would need to repeat the generic device information in their fields. Not sure, but it may be that it could be a good thing to prevent this repetition of generic device fields by having the same hierarchy in the value objects and DTOs. Given this is not how the code was organized this would **not** be something to be addressed in scope of this pull request I think.
To avoid misunderstanding, not every `Device` is a CDMA device, devices can be connected with other communication methods as well.